src/coordination/coordinator_instance.cpp

```diff
@@ -55,23 +56,41 @@ CoordinatorInstance::CoordinatorInstance()
                                            &CoordinatorInstance::MainFailCallback);
             });
 
-            std::ranges::for_each(repl_instances_, [this](auto &instance) {
-              instance.SetNewMainUUID(raft_state_.GetUUID());
-              instance.StartFrequentCheck();
-            });
+            std::ranges::for_each(repl_instances_, [](auto &instance) { instance.StartFrequentCheck(); });
+            is_shutting_down_ = false;
           },
           [this]() {
-            spdlog::info("Leader changed, stopping all replication instances!");
-            repl_instances_.clear();
+            thread_pool_.AddTask([this]() {
+              spdlog::info("Leader changed, trying to stop all replication instances frequent checks!");
+              is_shutting_down_ = true;
```

Comment:
Why is_shutting_down_? Plus this seems not synchronized

Reply:
Resolved as per your other comment

Reply:
I think we can discuss better name, but I was not sure